stdlib/experimental/prometheus/scrape.go

```diff
@@ -0,0 +1,334 @@
+	reader         *execute.RowReader
+
+	metrics []Metric
+	met     Metric
+	i       int
+	url     string
+	resp    *http.Response
+}
+
+// Metric stores the fields that we need to construct Table
+type Metric struct {
+	Name      string                 `json:"name"`
+	Tags      map[string]string      `json:"tags"`
+	Fields    map[string]interface{} `json:"fields"`
+	Timestamp time.Time              `json:"timestamp"`
+}
+
+func (p *PrometheusIterator) Connect(ctx context.Context) error {
+	p.url = p.spec.URL // for easier reference
+
+	resp, err := http.Get(p.url)
+	p.resp = resp
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Parse the entire body into a slice of Metrics
+	err = p.parse(resp.Body, resp.Header)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *PrometheusIterator) parse(r io.Reader, header http.Header) (err error) {
+	var parser expfmt.TextParser
+	now := time.Now()
+
+	mediatype, params, err := mime.ParseMediaType(header.Get("Content-Type"))
+	if err != nil {
+		return err
+	}
+
+	// Prepare output
+	metricFamilies := make(map[string]*dto.MetricFamily)
+	if mediatype == "application/vnd.google.protobuf" &&
+		params["encoding"] == "delimited" &&
+		params["proto"] == "io.prometheus.client.MetricFamily" {
+		for {
+			mf := &dto.MetricFamily{}
+			if _, err := pbutil.ReadDelimited(r, mf); err != nil {
+				if err == io.EOF {
+					break
+				}
+				return fmt.Errorf("reading metric family protocol buffer failed: %s", err)
+			}
+			metricFamilies[mf.GetName()] = mf
+		}
+	} else {
+		metricFamilies, err = parser.TextToMetricFamilies(r)
+		if err != nil {
+			return fmt.Errorf("reading text format failed: %s", err)
+		}
+	}
+
+	p.metrics = make([]Metric, 0)
+
+	// read metrics
+	for name, family := range metricFamilies {
+		for _, m := range family.Metric {
+			// read tags
+			tags := makeLabels(m)
+
+			// read fields
+			fields := make(map[string]interface{})
+			fields = getNameAndValue(m)
+
+			// Store data needed to build flux Table
+			if len(fields) > 0 {
+				tm := now
+				if m.TimestampMs != nil && *m.TimestampMs > 0 {
+					tm = time.Unix(0, *m.TimestampMs*1000000)
+				}
+				met := Metric{
+					Timestamp: tm,
+					Tags:      tags,
+					Fields:    fields,
+					Name:      name,
+				}
+				p.metrics = append(p.metrics, met)
+			}
+		}
+	}
+	return nil
+}
+
+func (p *PrometheusIterator) Fetch(ctx context.Context) (bool, error) {
+	// Iterate over all Metrics in the slice
+	p.i++
+	if p.i < len(p.metrics) {
+		p.met = p.metrics[p.i]
+		return true, nil
+	}
+
+	// No more metrics to return
+	return false, nil
+}
+
+func (p *PrometheusIterator) Decode(ctx context.Context) (table flux.Table, err error) {
+
+	// Unpacking Fields map
+	var val interface{}
+	for _, v := range p.met.Fields {
+		val = v
+	}
+
+	groupKey := execute.NewGroupKeyBuilder(nil)
+	groupKey.AddKeyValue("_measurement", values.New("prometheus"))
+	groupKey.AddKeyValue("_field", values.New(p.met.Name))
+
```

Comment:
At least in the metrics that we get when telegraf scrapes `/metrics` endpoints in acceptance and production, we usually make the `_measurement` be the prometheus metric name (like `http_query_request_bytes`), and `_field` is the metric type (like `counter` or `gauge`).

Probably we want to do the same thing?

Reply:
Oh, I like the idea of adding `url` as a column to the output, probably we want to put that in the group key as well?

Reply:
So I spoke with Daniel Nelson recently and he told that telegraf has changed their prometheus scraper to omit the `metric type`  put **prometheus** as the measurement column. I originally had it the way you suggested to match the chronograf example that you showed me.

Reply:
Ok, that's great.  Thanks for clarifying this.